### PR TITLE
CI: add pypi package check

### PR DESCRIPTION
**Pull request by orbeckst**
partially addresses #62

### CODE REVIEW

**Comment by orbeckst** [comment]
our CI is not running on PRs so mainly I want to find out why

**Comment by orbeckst** [comment]
@xiki-tempula could you have a quick look? I'd like to merge soon, mostly to ensure that we have CI running because at the moment, other PRs such as #71  don't run tests.

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -26,14 +26,23 @@ jobs:
         os: [macOS-latest, ubuntu-latest, windows-latest]
         python-version: ["2.7", "3.5", "3.6", "3.7", "3.8", "3.9", "3.10"]
```
Comment (xiki-tempula) [inline .github/workflows/CI.yaml:27]: python-version: ["3.7", "3.8", "3.9", "3.10"]

File: .github/workflows/CI.yaml [hunk 2]
```diff
@@ -78,3 +87,31 @@ jobs:
...
+    - name: Build alchemtest source dist
+      run: |
+        python setup.py sdist
+        
+    - name: Check package build sdist
+      run: |
+        DISTRIBUTION=$(ls -t1 dist/alchemtest-*.tar.gz | head -n 1)
+        test -n "${DISTRIBUTION}" || { echo "no distribution dist/alchemtest-*.tar.gz found"; exit 1; }
+        echo "twine check $DISTRIBUTION"
+        twine check $DISTRIBUTION
```
Comment (xiki-tempula) [inline .github/workflows/CI.yaml:117]: extra line。

File: .github/workflows/CI.yaml [hunk 3]
```diff
@@ -26,14 +26,23 @@ jobs:
...
         exclude:
-        - os: macOS-latest
-          python-version: 3.5	
         - os: macOS-latest
           python-version: 3.6
         - os: macOS-latest
           python-version: 3.7
         - os: macOS-latest
           python-version: 3.8
+        - os: macOS-latest
```
Comment (xiki-tempula) [inline .github/workflows/CI.yaml:35]: Why do we want to exclude all these versions? If we don't want to run on OSX and windows, we could just do
`os: [ubuntu-latest, ]`

**Review by xiki-tempula — CHANGES_REQUESTED** [review]
This would be a very good way of checking of the files are actually being included. Some questions.

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (orbeckst) [reply .github/workflows/CI.yaml:27]: We _can_ do NEP29 for alchemtest but to be honest, as long as it works I'm ok with just testing on supported platforms. 

If you want to follow NEP29, open an issue.

File: .github/workflows/CI.yaml (hunk 3 above)
Comment (orbeckst) [reply .github/workflows/CI.yaml:35]: To cut down on CI time and resources I only test lowest and highest Python on the windows and macOS runners.

Comment (xiki-tempula) [reply .github/workflows/CI.yaml:35]: I see the logic now. Might be good to add a comment to explain that. Otherwise, I'm happy with this PR.

**Review by xiki-tempula — APPROVED** [review]

**Comment by orbeckst** [comment]
> This would be a very good way of checking of the files are actually being included.

This PR does *not* really check that all files are included. It only checks that we can build the sdist.

We'd need to build the package, install, run tests for the installed package. Not very difficult but just more than I have time for right now.

**Pull request merged**